app/Http/Middleware/HasUserCheck.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use App\Contracts\InternalLycheeException;
+use App\Exceptions\UnauthorizedException;
+use App\Http\Middleware\Checks\IsInstalled;
+use Closure;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+
+class HasUserCheck
+{
+	private IsInstalled $isInstalled;
+
+	public function __construct(IsInstalled $isInstalled)
+	{
+		$this->isInstalled = $isInstalled;
+	}
+
+	/**
+	 * Handle an incoming request.
+	 *
+	 * @param Request $request
+	 * @param Closure $next
+	 *
+	 * @return mixed
+	 *
+	 * @throws UnauthorizedException
+	 * @throws InternalLycheeException
+	 */
+	public function handle(Request $request, Closure $next): mixed
+	{
+		if (!$this->isInstalled->assert()) {
+			return $next($request);
+		}
+
+		if (!Auth::hasUser()) {
+			throw new UnauthorizedException('Requires authentication');
+		}
+
+		return $next($request);
+	}
```

Comment:
What is the purpose of this middleware? What does this middleware ensure what is not already ensured by other middleware?

Reply:
> Otherwise, on the `api-admin` routes with disabled `AdminCheck` middleware, the endpoints are accessible without authorization.

This sounds like a bug somewhere else. In particular, all non-admin routes (e.g. `Photo::setTitle`) don't use the `AdminCheck` middleware, but still require authentication and these routes have done so for quite a while. I don't get why we suddenly need _another_ middleware to solve a problem which has already been solved many times for other routes without this new middleware.

> [Otherwise, ] This can lead to 500 errors because Auth::user() returns null.

Could you please give a specific example where and when this happens? This sounds like a bug in a particular route.

In particular, the only "old" routes which are inside the `api-admin` group for historic reasons, but explicitly exclude the `AdminCheck` middleware are `User::setEmail` and `User::getEmail`. If no user is authenticated (even no non-admin user) these routes throw an `UnauthenticatedException` and return a HTTP status code 401 which is just fine.

The only other routes which are inside the `api-admin` group, but exclude the `AdminCheck` middleware are the new routes `User::getAuthenticatedUser`, `User::resetToken` and `User::unsetToken` which are added bis this PR. If these routes generate a 500 Server Error then they are broken somehow and we should fix that instead of trying to work around it.

Reply:
Never mind. Fixed it myself in https://github.com/LycheeOrg/Lychee/pull/1368/commits/8e9cde160a630c275ac528bbd062d1bfbe51b33d.